### PR TITLE
Fixes #29608 - optimize encryption check

### PR DESCRIPTION
**Pull request by lzap**
Small micro optimalization, probably zero effect overall but that check
is terrible.

```
Warming up --------------------------------------
               slice   194.579k i/100ms
          start_with   408.562k i/100ms
               index   245.402k i/100ms
Calculating -------------------------------------
               slice      2.922M (± 1.6%) i/s -     14.788M in   5.062824s
          start_with     10.473M (± 1.1%) i/s -     52.704M in   5.033140s
               index      3.892M (± 0.9%) i/s -     19.632M in   5.044098s
```

### CODE REVIEW

**Review by ekohl — APPROVED** [review]
Just for readability this is already a win IMHO.

File: app/models/concerns/encrypt_value.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def is_encryptable?(str)
   end
 
   def is_decryptable?(str)
```
Comment (adamruzicka) [inline app/models/concerns/encrypt_value.rb:21]: How about we turn the entire method into this while we're at it
```ruby
!encryption_key.blank? && matches_prefix(str)
```

Comment (ekohl) [reply app/models/concerns/encrypt_value.rb:21]: Oh nice. Perhaps `is_encryptable` could follow the same pattern.

Comment (tbrisker) [reply app/models/concerns/encrypt_value.rb:21]: gotta love De Morgan's Laws :)

**Review by tbrisker — APPROVED** [review]
Thanks @lzap !

**Comment by tbrisker** [comment]
opened https://github.com/theforeman/foreman/pull/7625 for the condition refactoring.

**Pull request merged**